components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -945,42 +945,47 @@ func runIDEReadinessProbe(cfg *Config, ideConfig *IDEConfig, ide IDEKind) (deskt
 			port   = defaultIfZero(ideConfig.ReadinessProbe.HTTPProbe.Port, defaultProbePort)
 			url    = fmt.Sprintf("%s://%s:%d/%s", schema, host, port, strings.TrimPrefix(ideConfig.ReadinessProbe.HTTPProbe.Path, "/"))
 			client = http.Client{Timeout: 1 * time.Second}
-			tick   = time.NewTicker(500 * time.Millisecond)
+			tick   = time.NewTicker(250 * time.Millisecond)
 		)
 		defer tick.Stop()
 
 		t0 := time.Now()
 
-		for {
-			<-tick.C
-
+		for ; true; <-tick.C {
 			resp, err := client.Get(url)
 			if err != nil {
+				log.WithField("ide", ide.String()).WithError(err).Debug("Error running IDE readiness probe")
 				continue
 			}
-			defer resp.Body.Close()
 
-			if resp.StatusCode == http.StatusOK {
-				log.WithField("ide", ide.String()).WithField("status", resp.StatusCode).Infof("IDE readiness took %.3f seconds", time.Since(t0).Seconds())
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			resp.Body.Close()
+			if err != nil {
+				log.WithField("ide", ide.String()).WithError(err).Infof("Error reading response body from IDE status probe.")
+				break
+			}
 
-				if ide == DesktopIDE {
-					bodyBytes, err := ioutil.ReadAll(resp.Body)
-					log.WithField("ide", ide.String()).Infof("IDE status probe body: %s", string(bodyBytes))
-					if err != nil {
-						log.WithField("ide", ide.String()).WithError(err).Infof("Error reading response body from IDE status probe.")
-						break
-					}
-					err = json.Unmarshal(bodyBytes, &desktopIDEStatus)
-					if err != nil {
-						log.WithField("ide", ide.String()).WithError(err).WithField("body", bodyBytes).Debugf("Error parsing JSON body from IDE status probe.")
-						break
-					}
-					log.WithField("ide", ide.String()).Infof("Desktop IDE status: %s", desktopIDEStatus)
-				}
+			if resp.StatusCode != http.StatusOK {
+				log.WithField("ide", ide.String()).WithField("status", resp.StatusCode).Info("IDE readiness probe came back with non-200 status code")
+				continue
+			}
+
+			log.WithField("ide", ide.String()).WithField("status", resp.StatusCode).Infof("IDE readiness took %.3f seconds", time.Since(t0).Seconds())
+
+			if ide != DesktopIDE {
+				break
+			}
+
+			log.WithField("ide", ide.String()).Infof("IDE status probe body: %s", string(bodyBytes))
+
+			err = json.Unmarshal(bodyBytes, &desktopIDEStatus)
+			if err != nil {
+				log.WithField("ide", ide.String()).WithError(err).WithField("body", bodyBytes).Debugf("Error parsing JSON body from IDE status probe.")
 				break
 			}
 
-			log.WithField("ide", ide.String()).WithField("status", resp.StatusCode).Info("IDE readiness probe came back with non-200 status code")
+			log.WithField("ide", ide.String()).Infof("Desktop IDE status: %s", desktopIDEStatus)
+			break
```

Comment:
I think it would be better to move all this logic outside the loop so we don't have that many `breaks`

Reply:
fixed